runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/DistributionCell.java

```diff
@@ -52,6 +52,12 @@ public DistributionCell(MetricName name) {
     this.name = name;
   }
 
+  @Override
+  public void reset() {
+    dirty.afterModification();
```

Comment:
I don't believe reset() should make this dirty since reset() should set this to the state it is at as if it was newly constructed.